src/odemis/model/_components.py

```diff
@@ -107,12 +107,16 @@ class Component(ComponentBase):
     Component to be shared remotely
     '''
 
-    def __init__(self, name, parent=None, children=None, daemon=None):
+    def __init__(self, name, parent=None, children=None, dependencies=None, daemon=None):
         """
         name (string): unique name used to identify the component
         parent (Component): the parent of this component, that will be in .parent
         children (dict str -> Component): the children of this component, that will
-            be in .children. Objects not instance of Component are skipped
+            be in .children. Objects not instance of Component are skipped. This means that
+            this argument is only relevant for the microscope class since all other classes
+            don't have component children (they are dependencies).
+        dependencies (dict str -> Component): the dependencies of this component, that will
```

Comment:
The `dependencies` argument should only get Components. So I'd suggest to raise an Error is the objects are not Component (instead of silently skipping them).